src/main/java/uk/gov/pay/api/filter/RateLimiterFilter.java

```diff
@@ -43,31 +50,34 @@ public RateLimiterFilter(RateLimiter rateLimiter, ObjectMapper objectMapper) {
         this.objectMapper = objectMapper;
     }
 
-    @Override
-    public void init(FilterConfig filterConfig) {}
+    private void setTooManyRequestsError() throws IOException {
+        String errorResponse = objectMapper.writeValueAsString(anApiErrorResponse(Code.TOO_MANY_REQUESTS_ERROR));
+        Response.ResponseBuilder builder = Response.status(TOO_MANY_REQUESTS_STATUS_CODE)
+                .entity(errorResponse)
+                .encoding(UTF8_CHARACTER_ENCODING)
+                .variant(new Variant(MediaType.APPLICATION_JSON_TYPE, "", UTF8_CHARACTER_ENCODING));
 
-    @Override
-    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
+        throw new WebApplicationException(builder.build());
+    }
 
-        final String authorization = ((HttpServletRequest) request).getHeader("Authorization");
-        final String method = ((HttpServletRequest) request).getMethod();
+    @Override
+    public void filter(ContainerRequestContext requestContext) throws IOException {
+        final String authorization = requestContext.getHeaderString("Authorization");
+        final String method = requestContext.getMethod();
 
+        String accountId = getAccountId(requestContext);
         try {
-            rateLimiter.checkRateOf(method + "-" + authorization, method);
-            chain.doFilter(request, response);
+            rateLimiter.checkRateOf(accountId, method + "-" + authorization, method);
         } catch (RateLimitException e) {
-            LOGGER.info("Rate limit reached for current service. Sending response '429 Too Many Requests'");
-            setTooManyRequestsError((HttpServletResponse) response);
-        } 
+            LOGGER.info("Rate limit reached for current service and account [{}]. Sending response '429 Too Many Requests'", accountId);
```

Comment:
Looks like we check the rate limiting based on a composite key comprised of the method and the api key. Makes sense not to log the api key but should this log entry mention the method along with the account id? I see we log the method as part of throwing the `RateLimiteException` so perhaps not necessary to add it here too, so happy to approve as-is 👍

Reply:
Added `method` for consistency